interface/patient_file/deleter.php

```diff
@@ -409,7 +409,8 @@ function popup_close() {
     }
 
     if ($encounterid) { //this code need to be same as 'parent.imdeleted($encounterid)' when the popup is div like
-        echo "window.opener.imdeleted(" . attr($encounterid) . ");\n";
+        // convert all characters to HTML entities to prevent xss attack with additional parentheses
+        echo "window.opener.imdeleted(" . htmlentities($encounterid, ENT_HTML5) . ");\n";
     } else {
```

Comment:
Prevent attack like -
http://localhost/openemr/interface/patient_file/deleter.php?encounterid=9);alert(123);//

Reply:
I was not able to recreate this attack. The attr() in my testing protects from this vulnerability. Ensure you tested this on the most recent codebase.
It's best to avoid htmlentities since that can be tied to encoding and lead to issues in internationalization.